sources/HUBViewModelDiff.m

```diff
@@ -57,7 +64,9 @@ + (instancetype)diffFromViewModel:(id<HUBViewModel>)fromViewModel
                         algorithm:(HUBDiffAlgorithm)algorithm
 {
     NSParameterAssert(algorithm);
-    return algorithm(fromViewModel, toViewModel);
+    HUBViewModelDiff *diff =  algorithm(fromViewModel, toViewModel);
+    [diff calculateHeaderChangesFromViewModel:fromViewModel toViewModel:toViewModel];
+    return diff;
```

Comment:
My point is that the diffing functions are defined as diffing view models, not body components. That is, one call to the diffing function should return the _entire_ diff, including body components, overlays and header.

We could (and probably should) change this if we wanted to diff multiple lists of components, but then it should return something else than a HUB_View_ModelDiff.